neqo-transport/src/addr_valid.rs

```diff
@@ -419,9 +419,18 @@ impl NewTokenSender {
             if t.needs_sending && t.len() <= builder.remaining() {
                 t.needs_sending = false;
 
+                #[cfg(feature = "build-fuzzing-corpus")]
+                let frame_start = builder.len();
+
                 builder.encode_varint(FrameType::NewToken);
                 builder.encode_vvec(&t.token);
 
+                #[cfg(feature = "build-fuzzing-corpus")]
+                neqo_common::write_item_to_fuzzing_corpus(
+                    "frame",
+                    &builder.as_ref()[frame_start..],
+                );
```

Comment:
Given the general shape of this, would it make sense to define a function that you could call like this?

```rust
builder.encode_frame(FrameType::NewToken, |b| {
  b.encode_vvec(&t.token);
});
```

That wrapper could do the special fuzzing corpus logging, rather than have it spread everywhere.

You might need to make a crate-local trait to add the function implementation to the builder, rather than have that code in `neqo-common`.